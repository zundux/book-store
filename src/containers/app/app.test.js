import React from "react";
import ReactDOM from "react-dom";
import App from "./app";

describe("App container", () => {

  it("App renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("App container", () => {
    const component = shallow(<App/>);
    matchSnapshot(component);
  });
});