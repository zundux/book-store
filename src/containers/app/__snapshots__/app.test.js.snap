// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container App container 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ConnectedRouter)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <WithStyles(CssBaseline) />
    <header>
      <nav>
        <Link
          replace={false}
          to="/"
        >
          Catalog
        </Link>
        <Link
          replace={false}
          to="/cart"
        >
          Cart
        </Link>
        <Link
          replace={false}
          to="/about-us"
        >
          About
        </Link>
      </nav>
    </header>
    <main>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/product/:id"
        />
        <Route
          component={[Function]}
          path="/cart"
        />
        <Route
          component={[Function]}
          path="/order/created"
        />
        <Route
          component={[Function]}
          path="/order"
        />
        <Route
          component={[Function]}
          path="/about-us"
        />
        <Route
          component={[Function]}
        />
      </Switch>
    </main>
    <footer />
    <Modals />
  </Connect(ConnectedRouter)>
</Provider>
`;
